### ISSUE TITLE
⌘D(単語の置き換え)でIMEによる入力が反映されません

### ISSUE THREAD

**Issue by ykamei** [coteditor/CotEditor#1083]
こんにちは。素晴らしいアプリの開発とメンテナンスありがとうございます。
単語の置き換えで気づいた点をご報告させていただきます。

文字列が選択された状態で⌘Dを押すと同じ文字列が選択され、
そのまま文字を入力するとすべて置き換えされますが、
置き換えしたい文字列をIME経由で入力すると
一番最初に選択された文字列しか置き換えされません。

文字列例)
SUMMER_GIFT_01
SUMMER_GIFT_02
SUMMER_GIFT_03
SUMMER_GIFT_04

手順:
1. SUMMERを選択し、⌘Dを4回押す
2. IME経由で 夏 と入力しEnter
3. 結果

夏_GIFT_01
SUMMER_GIFT_02
SUMMER_GIFT_03
SUMMER_GIFT_04

期待する結果は

夏_GIFT_01
夏_GIFT_02
夏_GIFT_03
夏_GIFT_04

です。
ご確認お願いできますと幸いです。
宜しくお願いいたします。

**Comment by 1024jp** [comment]
本件は残念ながらバグではなく仕様となります。
現時点でCotEditorは日本での複数カーソル入力に対応しておりません。

<img width="800" alt="スクリーンショット 2020-06-19 14 58 01" src="https://user-images.githubusercontent.com/1165044/85101456-67a77b80-b23d-11ea-98ab-8e9a70164bbd.png">


これは、Appleが提供するテキスト入力のシステムの制限によるものです。
（Appleの技術者からも「できない」という回答をもらっています）
模擬的な迂回策は模索はしているのですが、なかなか実現は難しく、
すぐ改良できるというお約束はできず。
長期的な課題とさせてください。

**Comment by ykamei** [comment]
返信ありがとうございます。
仕様とのこと承知しました。ヘルプページの確認が不足しており申し訳ありません。
別ソフトウェアで恐縮ですが、Sublime Text3では実現できているのを確認したので
なにかトリッキーな解決方法があるのかもしれませんね。

**Comment by 1024jp** [comment]
> 別ソフトウェアで恐縮ですが、Sublime Text3では実現できているのを確認したので
> なにかトリッキーな解決方法があるのかもしれませんね。

Sublime Textはそもそも使っているフレームワークが違うのでこの制約を受けないのです...。

**Comment by ykamei** [comment]
フレームワークが違うとのこと理解しました。Appleの対応に期待したいところですね。

**Comment by bfishadow** [comment]
Chinese input has same issue. Looks like this is a CJK problem.

**Comment by 1024jp** [comment]
@bfishadow That’s right. This is a limitation common to all languages that require a two-step input sequence. I’ve updated the title to make it clearer.
